kep/numa-nodes-scoring/README.md

```diff
@@ -0,0 +1,297 @@
+			combinationResources := combineResources(numaNodes, combination)
+
+			resourcesFit := true
+                        hasNUMAResource := false
+			for resource, quantity := range resources {
+				if quantity.IsZero() {
+					// why bother? everything's fine from the perspective of this resource
+					klog.V(4).InfoS("ignoring zero-qty resource request", "logKey", logKey, "node", nodeName, "resource", resource)
+					continue
+				}
+
+				combinationQuantity, ok := combinationResources[resource]
+				if !ok {
+					// non NUMA resource continue
+					continue
+				}
+
+                                hasNUMAResource := true
+
+				if !isCombinationSuitable(qos, resource, quantity, combinationQuantity) {
+					resourcesFit = false
+					break
+				}
+
+			}
+                        // if pod uses only non NUMA resources return 0
+                        if !hasNUMAResource{
+                                return combinationBitmask, 0
+                        }
+
+			// if resources can be fit on given combination, just return the number of numa nodes requires to fit them
+			// according to TopologyManager if both masks are the same size pick the one that has less bits set
+			// https://github.com/kubernetes/kubernetes/blob/3e26e104bdf9d0dc3c4046d6350b93557c67f3f4/pkg/kubelet/cm/topologymanager/bitmask/bitmask.go#L146
+			// combin.Combinations is generating combinations in an order from the smallest to highest value
+			if resourcesFit {
+				combinationBitmask.Add(combination...)
+				return combinationBitmask, i
+			}
+		}
+	}
+
+	// score plugin should be running after resource filter plugin so we should always find sufficient amount of NUMA nodes
+	klog.V(1).InfoS("Shouldn't be here, no combination is able to satisfy resource requirements", "logKey", logKey, "node", nodeName)
+	return combinationBitmask, -1
+}
+
+// combineResources sums resources for given NUMA nodes combination
+func combineResources(numaNodes noderesourcetopology.NUMANodeList, combination []int) v1.ResourceList {
+	resources := v1.ResourceList{}
+	for _, nodeIndex := range combination {
+		for resource, quantity := range numaNodes[nodeIndex].Resources {
+			if value, ok := resources[resource]; ok {
+				value.Add(quantity)
+				resources[resource] = value
+				continue
+			}
+			resources[resource] = quantity
+		}
+	}
+
+	return resources
+}
+```
+
+
+### Description of substracting resources
+
+The Topology Manager can deal with the alignment of resources in a couple of distinct scopes:
+* container
+* pod
+
+Calculating required NUMA nodes for a `pod` scope is straightforward, we just need to calculate it for pod's effective resources.
+
+When it comes to `container` scope we need to calculate `required` NUMA nodes for every container in proper order and
+temporarly substract those resources from available resources. Return value would be the maximum amount of NUMA nodes required to run
+container in given pod.
+
+```go
```

Comment:
Ditto. I'd prefer using concise stanzas or pseudo code to explain the core logic.